engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/HashCodeNode.java

```diff
@@ -143,12 +143,16 @@ long hashCodeForBigInteger(EnsoBigInteger bigInteger) {
     return bigInteger.getValue().hashCode();
   }
 
-  @Specialization(guards = {"!interop.fitsInLong(v)", "interop.fitsInBigInteger(v)"})
+  @Specialization(guards = {"interop.isNumber(v)"})
```

Comment:
I am glad to see all the tests for new behavior while the only (_dangerous_, potentially _non compatible_) code change is here in `HashCodeNode` handling _small interop BigIntegers_. I'd like @Akirathan to take a look as I am:
- opening up to all `isNumber` values
- either treating them as `java.math.BigInteger` or falling back to `hashCodeForBigInteger`

I am not completely sure this is consistent with other `@Specialization`s in this class.

Reply:
The most attractive option I see right now:
- direct specializations for Enso number types - e.g. `long`, `double`, `EnsoBigInteger`
- having `interop.isNumber(v)` guarded specialization that uses a _convert to Enso value_ node and then computes the `hashCode`.
- removing all other number related (e.g. `int`, `byte`, etc.) specializations

Note: the _convert to Enso value_ node needs to also enforce conversion of `fitsInBigInteger` to `EnsoBigInteger`.

Reply:
I am not sure if I entirely understand the motivation here. First of all, I think that the current change introduced by you is potentially wrong. If `v` is, e.g., a primitive double, then it does not follow the same logic as in `hashCodeForDouble`.  But I can see that all the tests are passing, so we can merge this PR now and address this as soon as we have a failing test.

Reply:
> ...potentially wrong. If v is, e.g., a primitive double, then...

Thank you for taking a look, Pavel! @GregoryTravis, it may be safer to replace the guards with:
```java
@Specialization(guards = {"interop.fitsInLong(v) || interop.fitsInBigInteger(v)"})
```
to make sure `double` isn't accepted by this `@Specialization`. If the tests continue to pass, then it is better to _play it safe_.